qemu/tests/block_hotplug_passthrough.py

```diff
@@ -0,0 +1,91 @@
+import logging
+
+from virttest import error_context
+from virttest import utils_test
+from virttest import utils_disk
+from avocado.utils import process
+from provider.storage_benchmark import generate_instance
+from provider.block_devices_plug import BlockDevicesPlug
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Hotplug/unplug passthrough disk test:
+    1) Create passthrough disk with scsi_debug tool.
+    2) Start the guest.
+    3) Hotplug this passthrough disk.
+    4) Create partition on this disk and format it.
+    5) Do iozone/dd test on this disk.
+    6) Reboot the guest.
+    7) Unplug this passthrough disk.
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+
+    def create_path_disk():
+        """Create a passthrough disk with scsi_debug """
+        create_cmd = params.get("create_command")
+        process.system_output(create_cmd, shell=True, verbose=False)
+
+    def hotplug_path_disk(vm):
+        """Hotplug passthrough disk."""
+        error_context.context("Hotplug passthrough device", logging.info)
+        path_dev = process.getoutput("ls /dev/sd* | tail -n 1", shell=True)
+        vm.params["image_name_stg0"] = path_dev
+        plug = BlockDevicesPlug(vm)
+        plug.hotplug_devs_serial()
+
+    def format_plug_disk(session):
+        """Format new hotpluged disk."""
+        stg_image_size = params["stg_image_size"]
+        ostype = params["os_type"]
+        if ostype == "windows":
+            disk = utils_disk.get_windows_disks_index(session, stg_image_size)
+        else:
+            disk = sorted(utils_disk.get_linux_disks(session).keys())
+        partition = utils_disk.configure_empty_disk(session, disk[0],
+                                                    stg_image_size, ostype)
+        if not partition:
+            test.fail("Fail to format disks.")
+        else:
+            return partition[0]
+
+    def run_io_test(session, partition):
+        """ Run io test on the hot plugged disk. """
+        iozone_options = params.get('iozone_options', None)
+        dd_test = params.get('dd_test', None)
```

Comment:
You don't need to set `None` to `params.get('xxx', None)` since `def get(self, key, default=None):`

Reply:
updated, thanks~